py-polars/polars/io/delta.py

```diff
@@ -313,3 +315,114 @@ def _check_if_delta_available() -> None:
         raise ImportError(
             "deltalake is not installed. Please run `pip install deltalake>=0.9.0`."
         )
+
+
+def _reconstruct_field_type(
```

Comment:
Seems to me like this function should really be refactored, lots of nested if-else checks and there seems to be quite some duplication in this pattern. I find it hard to read. A helper function would probably be helpful?

Reply:
I'll have to take a look and rethink the logic. It's quite nested because I started with small statements and kept expanding.